packages/fluentui/react-northstar-prototypes/src/prototypes/VirtualizedStickyTree/VirtualStickyTree.tsx

```diff
@@ -0,0 +1,368 @@
+const getStickyItemStyle = (indexAmoungStickyItems: number, stickyItemNums: number, stickyItemSize: number) => ({
+  height: stickyItemSize,
+  zIndex: teamsTheme.siteVariables.zIndexes.overlay,
+  position: 'sticky',
+  top: indexAmoungStickyItems * stickyItemSize,
+  bottom: (stickyItemNums - indexAmoungStickyItems - 1) * stickyItemSize,
+  backgroundColor: teamsTheme.siteVariables.colorScheme.default.background3,
+});
+
+const InnerElementType = ({ children, style }, ref) => {
+  const context = React.useContext(InnerElementContext);
+  const { stickyItemIds, stickyItemPusherHeights, stickyItemSize, getItemById } = context;
+
+  const renderContent = React.useCallback(
+    (virtualItems: React.ReactElement<ListChildComponentProps>[]) => {
+      const createTreeItem = virtualItems[0]?.props.data?.createTreeItem;
+      if (!createTreeItem) {
+        return null;
+      }
+
+      const result: Record<
+        string,
+        {
+          stickyItem: React.ReactElement; // the sticky item itself
+          pusher: React.ReactElement; // the div pusher with height being the same as all descendents of this sticky item
+          children: React.ReactElement[]; // all descendents of this sticky item
+        }
+      > = {};
+
+      stickyItemIds.forEach((id, index) => {
+        result[id] = {
+          stickyItem: createTreeItem(id, getStickyItemStyle(index, stickyItemIds.length, stickyItemSize)),
+          pusher: (
+            <div
+              key={`${id}-pusher`}
+              style={{ height: stickyItemPusherHeights[index], zIndex: -1 }}
+              role="presentation"
+            />
+          ),
+          children: [],
+        };
+      });
+
+      virtualItems.forEach(virtualItem => {
+        const virtualItemId = virtualItem.key as string; // our `getItemKey` makes virtual item's key the same as its corresponding tree item's id
+        // get the sticky id to which the current virtualItem belongs to
+        let parentId = getItemById(virtualItemId)?.parent;
+        let parentItem = getItemById(parentId);
+        while (parentItem && parentItem.level > 1) {
+          parentId = parentItem.parent;
+          parentItem = getItemById(parentId);
+        }
+        if (result[parentId] == null) {
+          return;
+        }
+        result[parentId].children.push(virtualItem);
+      });
+
+      const flattenedResult = [];
+      stickyItemIds.forEach(id => {
+        flattenedResult.push(result[id].stickyItem);
+        flattenedResult.push(result[id].pusher);
+        result[id].children.forEach(child => {
+          flattenedResult.push(child);
+        });
+      });
+
+      return flattenedResult;
+    },
+    [getItemById, stickyItemIds, stickyItemPusherHeights, stickyItemSize],
+  );
+
+  return (
+    <div style={style} role="none">
+      {renderContent(children)}
+    </div>
+  );
+};
+
+const OuterElementType = React.forwardRef<HTMLDivElement>((props, ref) => <div ref={ref} {...props} role="none" />);
```

Comment:
nit: same here is the role necessary ? 
```suggestion
const OuterElementType = React.forwardRef<HTMLDivElement>((props, ref) => <div ref={ref} {...props} />);
```

Reply:
Yes, same as the previous comment